tests/ha/test_server_pod_failure.py

```diff
@@ -221,46 +198,27 @@ def test_degraded_reads_safe_server_pod_shutdown(self):
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Step 2: Performed READs and verified DI on the written data")
 
-        LOGGER.info("Step 3: Shutdown the server pod safely by making replicas=0")
-        LOGGER.info("Get server pod name to be deleted")
-        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
-        pod_name = random.sample(pod_list, 1)[0]
-        hostname = self.node_master_list[0].get_pod_hostname(pod_name=pod_name)
-        LOGGER.info("Deleting server pod %s", pod_name)
-        resp = self.node_master_list[0].create_pod_replicas(num_replica=0, pod_name=pod_name)
-        assert_utils.assert_false(resp[0], f"Failed to delete pod {pod_name} by making replicas=0")
-        LOGGER.info("Step 3: Successfully shutdown/deleted server pod %s by making replicas=0",
-                    pod_name)
-        self.deployment_name = resp[1]
+        LOGGER.info("Step 3: Shutdown random server pod safely by making replicas=0 and verify "
+                    "cluster & remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX])
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
```

Comment:
asserting on resp[1] shudn't it be reap[0] ??

Reply:
> asserting on resp[1] shudn't it be reap[0] ??

asserting if dict is empty .. Incase of failure in delete/shutdown pod. Hence resp[1]